tracing-core/src/field.rs

```diff
@@ -225,6 +225,14 @@ pub trait Visit {
 
     /// Visit a value implementing `fmt::Debug`.
     fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug);
+
+    /// Visit an Option
+    fn record_option(&mut self, field: &Field, value: Option<&(dyn Value + 'static)>) {
+        if let Some(inner_value) = value {
+            self.record_debug(field, &inner_value)
```

Comment:
I don't think this implements the recording of the interior type using the appropriate type-specific `record_*` method as described in the linked task.

You probably want to call the `record` method on the `inner_value`.

Reply:
Yup, that's right.

Reply:
Is this right `inner_value.record(field, self)`?

Reply:
> Is this right `inner_value.record(field, self)`?

Yeah, that's right.